modules/imapauth/lib/Auth/Source/MyAuth.php

```diff
@@ -0,0 +1,173 @@
+<?php
+class sspmod_imapauth_Auth_Source_MyAuth extends sspmod_core_Auth_UserPassBase {
```

Comment:
Don't worry, it's a very typical error among those who use namespaces for the first time 😉 

When you declare a namespace with the `namespace` keyword, that namespace applies to the entire file, not only to classes or interfaces defined there. Therefore, all the names used or declared in the file are relative to that namespace. The concept is quite similar to a file system. You may have a file in the root of your filesystem, named *file*. However, if you change your current directory to somewhere else with `cd /some/path`, then *file* is not in there of course, and to access it you need to give an absolute path: */file*.

The `namespace` keyword is equivalent to  `cd` in this context. So once you use it, every class, interface or other name used in that file needs to use either a relative *path* to it, or an *absolute* one. Since you are specifying a relative *path*, that's why the autoloader is trying to find `SimpleSAML\Module\unixauth\Auth\Source\sspmod_core_Auth_UserPassBase`. Just prepend those names with a `\`:

```php
<?php
namespace SimpleSAML\Module\unixauth\Auth\Source;

class UNIX extends \sspmod_core_Auth_UserPassBase {
```